### ISSUE TITLE
E221 False Positive

### ISSUE THREAD

**Issue by A-UNDERSCORE-D** [PyCQA/pycodestyle#1008]
E221 ("Multiple spaces before operator") is being seen in the below snippet on the -1:
```python
{
    "some long key":        1,
    "some other long key":  2,
    "short key":            3,
    "other key but neg":    -1,
}
```

I'm not sure if this is actually inconsistent, `-` isnt really being an operator there, or at the very least, as a unary one. Regardless its slightly inconsistent as the rest of those are fine, and it causes `autopep8` to automatically "fix" it by dropping consistency.

Edit: Additionally, it does not appear that a noqa silences this. I'd really prefer not to simply silence the entire rule because everywhere else its exactly what I'd want.

**Comment by Estartu** [comment]
-1 is a negativ value and not an operator.  There is no violation reporting it as such is a bug

**Comment by asottile** [comment]
it is a unary operator:

```console
$ python3 -c 'import ast; print(ast.dump(ast.parse("-1").body[0]))'
Expr(value=UnaryOp(op=USub(), operand=Constant(value=1, kind=None)))
```

**Comment by Estartu** [comment]
does the no multiple space rule apply to unary oprators or is the rule just for binary oprators. 
-    1 producese no warning while  2 -     1 generates a E222 multiple spaces after operator warning. 

so there is a difference between a unary operator and a binary operator. And the same difference should be hat E221

**Comment by asottile** [comment]
that's probably just an oversight in E222

**Comment by Estartu** [comment]
back to the original Problem. 

Why does a negativ  value in a formated dict produce a warning? Is there a violation to the intentions of pep8?

To quote from PEP8 
"A style guide is about consistency."

Is it consitent that positive integers can have multiple spaces in a dictionary and negative integers can not. I don't think so.

**Comment by asottile** [comment]
pycodestyle is fairly consistent that all of those lines are bad:

```
t2.py:2:21: E241 multiple spaces after ':'
t2.py:3:27: E241 multiple spaces after ':'
t2.py:4:17: E241 multiple spaces after ':'
t2.py:5:25: E241 multiple spaces after ':'
t2.py:5:25: E221 multiple spaces before operator
```

**Comment by Estartu** [comment]
no it's not

```
test = {'test':    1,
        'test123': 'test',
        'hurz':    -1
        }

test2 = -   1
test3 = 2 -   1
```
% pycodestyle '/Users/estartu/Library/Application Support/JetBrains/PyCharm2022.1/scratches/scratch.py'
/Users/estartu/Library/Application Support/JetBrains/PyCharm2022.1/scratches/scratch.py:3:16: E221 multiple spaces before operator
/Users/estartu/Library/Application Support/JetBrains/PyCharm2022.1/scratches/scratch.py:8:12: E222 multiple spaces after operator

**Comment by asottile** [comment]
```
t2.py:1:16: E241 multiple spaces after ':'
t2.py:3:16: E241 multiple spaces after ':'
t2.py:3:16: E221 multiple spaces before operator
t2.py:7:12: E222 multiple spaces after operator
```

**Comment by Estartu** [comment]
E241 is deactivated by default because most think it's no improvement and it's not enforced by PEP8. 
If the is a sereate error code for multiple spaces before an unary operator in could deactivate it for my projects. But there is only E221 and I think multiple spaces bevor an binary operator is bad so i don't want to deactiavte E221.

**Comment by Prometheus3375** [comment]
Any update on this issue?

**Comment by asottile** [comment]
looks like the conclusion is not a bug

**Comment by Prometheus3375** [comment]
pycodestyle is used by various tools, PyCharm is one of them. Inside PyCharm there is a setting to align dict literals on value. For this lines error E241 is suppressed, but not error E221. Support team asked to check behavior of pycodestyle, so this issue was created.

The problem is that the same space sequence reported twice. If I use a variable,
```python
negative = -1
{
    "some long key":        1,
    "some other long key":  2,
    "short key":            3,
    "other key but neg":    negative,
}
```
then there will be only error E241. If I inline `negative`, then E221 will be added.

---

One more insight from the example provided by @Estartu.

```python
test = {'test':    1,
        'test123': 'test',
        'hurz':    -1
        }

test2 = -   1
test3 = 2 -   1
```
I've just installed pycodestyle to check the behavior:
![image](https://github.com/PyCQA/pycodestyle/assets/35541026/6804ba05-9eab-421d-9893-545670088201)

E241 is not reported at all as well as spaces after minus at line 8. This is weird.

**Comment by asottile** [comment]
sounds like pycharm should also suppress the other error too -- this isn't a pycodestyle issue

**Comment by FichteFoll** [comment]
I agree with @Prometheus3375 on this one.

With the ever recurring example file:
```py
test = {'test':    1,
        'test123': 'test',
        'hurz':    -1
        }

test2 = -   1
test3 = 2 -   1
```
pycodestyle currently reports the following:
```
/tmp ∀ pycodestyle --select=E,W test.py
test.py:1:16: E241 multiple spaces after ':'
test.py:3:16: E221 multiple spaces before operator
test.py:3:16: E241 multiple spaces after ':'
test.py:7:12: E222 multiple spaces after operator
 ?=1
```
where I believe it should report this:
```
/tmp ∀ pycodestyle --select=E,W test.py
test.py:1:16: E241 multiple spaces after ':'
test.py:3:16: E241 multiple spaces after ':'
test.py:6:10: E222 multiple spaces after operator
test.py:7:12: E222 multiple spaces after operator
 ?=1
```
